File: Supervised Learning/Regression/Linear Regression/coding.py
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error


df = fetch_california_housing()

X = df.data
y = df.target

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42
)

model = LinearRegression()
model.fit(X_train, y_train)
y_predict = model.predict(X_test)

r2 = r2_score(y_test, y_predict)
mae = mean_absolute_error(y_test, y_predict)
mse = mean_squared_error(y_test, y_predict)
rmse = mean_squared_error(y_test, y_predict, squared=False)

print("R-squared score:", r2)
print("Mean Absolute Error:", mae)
print("Mean Squared Error:", mse)
print("Root Mean Squared Error:", rmse)

File: Supervised Learning/Regression/Linear Regression/coding.R
data("mtcars")

colnames(mtcars)

set.seed(42)

train_indices <- sample(1:nrow(mtcars), 0.8 * nrow(mtcars))
train_data <- mtcars[train_indices, ]
test_data <- mtcars[-train_indices, ]

model <- lm(mpg ~ ., data = train_data)

prediction <- predict(model, newdata = test_data)

r_squared <- summary(model)$r.squared
mse <- mean((test_data$mpg - prediction) ^ 2)
mae <- mean(abs(test_data$mpg - prediction))
rmse <- sqrt(mse)

print(paste("Mean Squared Error:", mse))
print(paste("Root Mean Squared Error:", rmse))
print(paste("Mean Absolute Error:", mae))
print(paste("R-squared:", r_squared))